### PR TITLE
add image_kwargs to report

### PR DESCRIPTION
**Pull request by SophieHerbst**
Responds to issue #848

Now that mne has been updated, we can pass image_kwargs to report.add_epochs. 
It works when I set the limits to the rejection limits in the config, but I cannot get it right when setting image_kwargs to None as per default. 

```
 File "/home/sh254795/Documents/REPOS/mne-study-template/mne_bids_pipeline/_config_import.py", line 340, in _default_factory
    idx = allowlist.index(val)
ValueError: {'image_kwargs': None} is not in list

    assert val == typ(), (key, val)
AssertionError: ('report_add_epochs_kwargs', {'image_kwargs': None})
```

### CODE REVIEW

File: mne_bids_pipeline/_config.py [hunk 1]
```diff
@@ -2131,6 +2131,20 @@ def noise_cov(bids_path):
     ```
 """
 
+report_add_epochs_image_kwargs: Optional[dict] = dict(image_kwargs=None)
```
Comment (larsoner) [inline mne_bids_pipeline/_config.py:2134]:
```suggestion
report_add_epochs_image_kwargs: Optional[dict] = dict()
```

File: mne_bids_pipeline/_config.py [hunk 2]
```diff
@@ -2131,6 +2131,20 @@ def noise_cov(bids_path):
...
+Specifies the limits for the color scales of the epochs_image in the report.
+If `None`, it defaults to the current default in MNE-Python.
+
+???+ example "Example"
+    Set vmin and vmax to the epochs rejection thresholds:
+    ```python
+    report_add_epochs_kwargs = dict(image_kwargs=dict(
+        grad = dict(vmin=-reject['grad'], vmax=reject['grad']), 
+        mag=dict(vmin=-reject['mag'],vmax=reject['mag'])))
+    ```
```
Comment (larsoner) [inline mne_bids_pipeline/_config.py:2145]: I would expect:
```suggestion
    report_add_epochs_kwargs = dict(
        grad = dict(vmin=-reject['grad'], vmax=reject['grad']), 
        mag=dict(vmin=-reject['mag'],vmax=reject['mag']))
    ```
```

File: mne_bids_pipeline/steps/preprocessing/_07_make_epochs.py [hunk 3]
```diff
@@ -256,6 +256,7 @@ def run_epochs(
             psd=psd,
             drop_log_ignore=(),
             replace=True,
+            image_kwargs=cfg.report_add_epochs_image_kwargs["image_kwargs"],
```
Comment (larsoner) [inline mne_bids_pipeline/steps/preprocessing/_07_make_epochs.py:259]:
```suggestion
            image_kwargs=cfg.report_add_epochs_image_kwargs,
```

File: mne_bids_pipeline/_config.py (hunk 1 above)
Comment (hoechenberger) [reply mne_bids_pipeline/_config.py:2134]: Should be `None`, yes

File: mne_bids_pipeline/_config.py (hunk 2 above)
Comment (hoechenberger) [reply mne_bids_pipeline/_config.py:2145]: Why the love for `dict`? I think `{}` is much easier to read and more versatile:

```python
    report_add_epochs_kwargs = {
        "grad": {
            "vmin": -reject["grad"],
            "vmax": reject["grad"]
        }, 
        "mag": {
            "vmin": -reject["mag"],
            "vmax": reject["mag"]
        }
    }
```

**Comment by SophieHerbst** [comment]
Also, it turns out that in order to use the rejection criteria from my config, I had to deal with the different scaling between rejection and plotting commands:
# # add image kwargs for report
# report_add_epochs_image_kwargs = {
#     "grad": {"vmin": 0, "vmax": reject["grad"]*1e15},
#     "mag": {"vmin": 0, "vmax": reject["mag"]*1e15},
# }

**Comment by larsoner** [comment]
For the consistency we should merge https://github.com/mne-tools/mne-bids-pipeline/pull/863 first probably, it should fix things for you I think

We should maintain compat with latest MNE stable, so you'll need to `inspect.getfullargspec(report.add_epochs)` to determine whether or not to set `image_kwargs` I think

**Comment by SophieHerbst** [comment]
I think I need some help here

**Review by larsoner — APPROVED** [review]
@SophieHerbst pushed a few commits to DRY (don't repeat yourself) the code a bit, and also test that the option works -- locally the report seems to reflect `reject`:

image

Can you look and make sure you're still happy?

**Comment by SophieHerbst** [comment]
Thank you @larsoner I ran it again on my data and I am very happy with it now.

**Comment by hoechenberger** [comment]
Thanks @SophieHerbst and @larsoner!!!

**Pull request merged**